cura/PrinterOutput/PrintJobOutputModel.py

```diff
@@ -29,6 +36,48 @@ def __init__(self, output_controller: "PrinterOutputController", key: str = "",
         self._assigned_printer = None  # type: Optional[PrinterOutputModel]
         self._owner = ""  # Who started/owns the print job?
 
+        self._configuration = None  # type: Optional[ConfigurationModel]
+        self._compatible_machine_families = []  # type: List[str]
+        self._preview_image_id = 0
+
+        self._preview_image = None
+
+    @pyqtProperty("QStringList", notify=compatibleMachineFamiliesChanged)
+    def compatibleMachineFamilies(self):
+        # Hack; Some versions of cluster will return a family more than once...
+        return set(self._compatible_machine_families)
+
+    def setCompatibleMachineFamilies(self, compatible_machine_families: List[str]) -> None:
+        if self._compatible_machine_families != compatible_machine_families:
+            self._compatible_machine_families = compatible_machine_families
+            self.compatibleMachineFamiliesChanged.emit()
+
+    @pyqtProperty(QUrl, notify=previewImageChanged)
+    def previewImageUrl(self):
+        self._preview_image_id += 1
+        # There is an image provider that is called "camera". In order to ensure that the image qml object, that
+        # requires a QUrl to function, updates correctly we add an increasing number. This causes to see the QUrl
+        # as new (instead of relying on cached version and thus forces an update.
+        temp = "image://print_job_preview/" + str(self._preview_image_id) + "/" + self._key
+        return QUrl(temp, QUrl.TolerantMode)
+
+    def getPreviewImage(self):
+        return self._preview_image
+
+    def updatePreviewImage(self, preview_image: Optional[QImage]):
+        if self._preview_image != preview_image:
+            self._preview_image = preview_image
+            self.previewImageChanged.emit()
+
+    @pyqtProperty(QObject, notify=configurationChanged)
+    def configuration(self) -> Optional["ConfigurationModel"]:
+        return self._configuration
+
+    def updateConfiguration(self, configuration: Optional["ConfigurationModel"]) -> None:
```

Comment:
This method is really a setter not an updater. It hangs on to the config which is given to it. It doesn't really update itself based the config.

Reply:
It is an updater, since set<something> is used to actually call the linked device. So if you want to update the info (which is what this does), updateConfiguration is required.

Reply:
I am not going to change this one.

I don't think the semantics are that important here. The question of overwrite vs. merging data isn't really addressed (we don't have two functions to handle them differently), and replacing a value can be a form of updating. Plus I think its nice that the updaters/setters all follow the same format (makes it easy to guess what the function is called).